src/dodal/cli.py

```diff
@@ -0,0 +1,55 @@
+import importlib
+
+import click
+
+from dodal.beamlines import ALL_BEAMLINES
+from dodal.beamlines.beamline_utils import (
+    DEFAULT_CONNECTION_TIMEOUT,
+    wait_for_connection,
+)
+from dodal.utils import make_all_devices
+
+from . import __version__
+
+
+@click.group(invoke_without_command=True)
+@click.version_option(version=__version__, message="%(version)s")
+@click.pass_context
+def main(ctx: click.Context) -> None:
+    if ctx.invoked_subcommand is None:
+        print("Please invoke subcommand!")
+
+
+@main.command(name="connect")
+@click.argument("beamline", type=click.Choice(list(ALL_BEAMLINES)), required=True)
```

Comment:
Passing the beamline module in as an argument may be confusing in cases where it conflicts with the `${BEAMLINE}` environment variable. However, in some cases the two are intentionally different. For example if running against s03 you load the `i03` beamline module but with `BEAMLINE=s03`, so `${BEAMLINE}` and the beamline module are two separate things with the same name and the same value in lots of cases, which seems a bit confusing. Not sure if it's worth tackling here or raising another issue.

Reply:
Could we default to using `${BEAMLINE}` (which will be the case for 90%) of our users then bomb out if it's not set?

Reply:
Yh, you're right, let's just require it

Reply:
I think it's still confusing given https://github.com/DiamondLightSource/dodal/pull/519#issuecomment-2107575051

Some other options are:
- Completely remove the `${BEAMLINE}` variable from dodal, base everything on the beamline module
- Have s03 as some kind of profile for i03 rather than base on the environment variable so you could do something like `dodal connect --profile tickit i03`

Reply:
Agreed, we then just move to solution 2 eventually as part of #502

Reply:
@DominicOram I think I've probably misunderstood this because what I've done is the same thing again. I made it so you can type:

```shell
dodal connect i04-1
```

which will try to connect to `s04-1` or

```shell
export BEAMLINE=i04-1
dodal connect
```

which try to connect to i04. That feels about as confusing and probably not what you had in mind.

Reply:
Have just thought of another requirement: the training rigs. They are all identical so should have one dodal module (I believe @paula-mg has prototyped one) and rely on `${BEAMLINE}` or similar to differentiate.

Reply:
Sorry, I wasn't clear my suggestion is literally to do `os.environ["BEAMLINE"] = beamline` inside `connect`

Reply:
Sounds good, also works for the training rig case